client/src/Components/Navigation/Header.js

```diff
@@ -1,28 +1,28 @@
-import React, { useEffect, useState } from "react";
-import { useDispatch } from "react-redux";
-import { useHistory, Link, useLocation } from "react-router-dom";
-import decode from "jwt-decode";
+import React, { useEffect, useState } from 'react'
+import { useDispatch } from 'react-redux'
+import { useHistory, Link, useLocation } from 'react-router-dom'
+import decode from 'jwt-decode';
 
 const Header = () => {
-  const dispatch = useDispatch();
-  const history = useHistory();
-  const location = useLocation();
-  const [user, setUser] = useState(JSON.parse(localStorage.getItem("profile")));
+  const dispatch = useDispatch()
+  const history = useHistory()
+  const location = useLocation()
+  const [user, setUser] = useState(JSON.parse(localStorage.getItem('profile')))
   useEffect(() => {
-    const token = user?.token;
+    const token = user?.token
     if (token) {
-      const decodedToken = decode(token);
+      const decodedToken = decode(token)
       const isTokenExpired = decodedToken.exp * 1000 < new Date().getTime()
-      if (isTokenExpired) logout();
+      if (isTokenExpired) logout()
     }
-    setUser(JSON.parse(localStorage.getItem("profile")));
-  }, [location]);
+    setUser(JSON.parse(localStorage.getItem('profile')))
```

Comment:
Same comment as above. Use the local storage keys constant for anywhere we are doing `localStorage.getItem` or `localStorage.setItem`